### PR TITLE
React Todo List 

### PR DESCRIPTION
**Pull request by kensteph**
In this PR I : 
- Create all the components
- Implement  CSS for the components
- Add functionalities : add, edit, complete, and delete todo
- Preserve the React state in the browser's local storage
- Implement Routing with React Router

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React</title>
```
Comment (biruk-bereka) [inline index.html]: [Optional]
- Using a template title is not recommended. Kindly change the Title of your document to a title that best describes your project.

File: src/components/TodoList/TodoList.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  // Add new Todo
+  const handleClickAddTodo = (title) => {
+    const newTodo = { id: uuidv4(), title, completed: false };
+    setTodos((prevState) => [...prevState, newTodo]);
+  };
+
+  // Remove Todo
+  const handleRemoveTodo = (id) => {
+    setTodos(todos.filter((todo) => todo.id !== id));
+  };
```
Comment (biruk-bereka) [inline src/components/TodoList/TodoList.jsx]: [Optional] 
- While using a set method of useState() it is recommended to use the previous State while setting the new state, in your case, kindly set the todos using the prevstate. 
```
setTodos((prevState)=>{prevState.filter((todo) => todo.id !== id))})
```

**Review by biruk-bereka — APPROVED** [review]
# Status Approved 💯

Hi @kensteph 

I am your reviewer @biruk-bereka

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉
## To Highlight 

-  Highly organized code. ✔️
-  No linter errors. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.
 
You can consider:
   - Pull request is made from the `feature branch` into the `dev` branch.
   
Cheers and Happy coding 👏🏽👏🏽

File: src/components/TodoList/TodoList.jsx (hunk 2 above)
Comment (kensteph) [reply src/components/TodoList/TodoList.jsx]: Thanks.

**Pull request merged**